subsys/net/lib/shell/net_shell_private.h

```diff
@@ -8,20 +8,40 @@
 #include <zephyr/shell/shell.h>
 #include <zephyr/net/net_ip.h>
 
-#define PR(fmt, ...)						\
-	shell_fprintf(sh, SHELL_NORMAL, fmt, ##__VA_ARGS__)
+#define PR(fmt, ...)                                                                               \
+	do {                                                                                       \
+		if (sh) {                                                                          \
+			shell_fprintf(sh, SHELL_NORMAL, fmt, ##__VA_ARGS__);                       \
+		}                                                                                  \
```

Comment:
For the else case, please add `printk(fmt, ##__VA_ARGS__);` per option 2 in the issue discussion (this is an update from the initial discussion).

Same applies to the other PR* macros in the rest of the commit.

Reply:
There is way too much back and forth with this simple PR and it seems to be stuck atm. In order to proceed lets add the `printk()` as @EricNRS  suggested above and lets beautify the `\` as I suggested so that there is not so much empty space and the line is shorter. This is max. 10 min work and then we can move on to more difficult tasks.

Reply:
Following @jukkar's strong recommendation to incorporate a check for `sh being NULL` and to fallback to `printk` within the `PR_*` macro, rather than exploring alternative options, I've decided to proceed with this approach. While I had hoped for a different solution, I recognize the importance of moving forward on this matter.

---
Ping @EricNRS / @trantanen if you have any response about this approach.

Reply:
Yes, this is fine.  It is just restoring it back to the original version before the cleanup.

Reply:
I'd probably add the setting of shell pointer as in https://github.com/zephyrproject-rtos/zephyr/pull/68812#issuecomment-1959466786 to make it work better in many cases.
Memory penalty for the printk seems to be a little high...

But if you need to keep it as it is, then that's it 😄

Reply:
@trantanen, I share your views and am keen to see your proposed solutions in another PR.
Looking forward to it! Regards.